pkg/serviceaccount/tokens_controller.go

```diff
@@ -270,6 +282,21 @@ func (e *TokensController) createSecretIfNeeded(serviceAccount *api.ServiceAccou
 
 // createSecret creates a secret of type ServiceAccountToken for the given ServiceAccount
 func (e *TokensController) createSecret(serviceAccount *api.ServiceAccount) error {
+	// We don't want to update the cache's copy of the service account
+	// so add the secret to a freshly retrieved copy of the service account
```

Comment:
If 2 is a concern, perhaps you should be using the work queue I added for endpoints controller.